Sources/ChessKit/Board.swift

```diff
@@ -136,19 +141,26 @@ public struct Board: Sendable {
                 ),
                 in: previousSet
             )
-
+            
             if updatedPiece.kind == .pawn {
                 position.resetHalfmoveClock()
-
+                
                 if abs(start.rank.value - end.rank.value) == 2 {
                     position.enPassant = EnPassant(pawn: updatedPiece)
+                    
+                    if validateEnPassant() {
+                        position.enPassantIsPossible = true
+                    }
+                    
                 }
             }
-
             return process(move: move)
         }
     }
-
+    
+    public func hashedPositions() -> [Int:Int] {
+        return self.positionHashCounts
+    }
```

Comment:
I don't think we need this function, see my suggestion for the tests so we don't need to expose `positionHashCounts` publicly.